File: src/No1022_从根到叶的二进制数之和/Solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def sumRootToLeaf(self, root: TreeNode) -> int:
        return self.dfsSum(root, 0)

    def dfsSum(self, root: TreeNode, sum) -> int:
        if not root:
            return 0
        sum = (sum << 1) + root.val
        if root.left is None and root.right is None:
            return sum
        return self.dfsSum(root.left, sum) + self.dfsSum(root.right, sum)

File: src/No1022_从根到叶的二进制数之和/Solution.go
package No1022_从根到叶的二进制数之和

//Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sumRootToLeaf(root *TreeNode) int {
	return dfsSum(root, 0)
}
func dfsSum(root *TreeNode, sum int) int {
	if root == nil {
		return 0
	}
	sum = (sum << 1) + root.Val
	if root.Left == nil && root.Right == nil {
		return sum
	}
	return dfsSum(root.Left, sum) + dfsSum(root.Right, sum)
}
